### PR TITLE
Pt 163760521 merge master to minerva

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/163760521

NOTE: Merge this, no squeezing or rebasing.

### CODE REVIEW

File: apps/aechannel/src/aesc_fsm.erl [hunk 1]
```diff
@@ -625,82 +642,112 @@ awaiting_signature(cast, {?SIGNED, slash_tx, SignedTx} = Msg,
...
+        fun() ->
+            D1 = send_shutdown_msg(SignedTx, D),
+            D2 = D1#data{latest = {shutdown, SignedTx},
+                        log    = log_msg(rcv, ?SIGNED, Msg, D1#data.log)},
+            next_state(mutual_closing, D2)
+        end, D);
 awaiting_signature(cast, {?SIGNED, ?SHUTDOWN_ACK, SignedTx} = Msg,
                    #data{latest = {sign, ?SHUTDOWN_ACK, CMTx}} = D) ->
     NewSignedTx = aetx_sign:add_signatures(
                     CMTx, aetx_sign:signatures(SignedTx)),
```
Comment (hanssv) [inline apps/aechannel/src/aesc_fsm.erl]: My eyes hurt from all the missing abstraction in this code, but that has nothing to do with the merge...

File: apps/aechannel/test/aesc_txs_SUITE.erl [hunk 2]
```diff
@@ -3412,7 +3412,7 @@ fp_register_name(Cfg) ->
             ?TEST_LOG("Contract create tx ~p", [ContractCreateTx]),
             OnChainTrees = aesc_test_utils:trees(S0),
             TxEnv = tx_env(#{height => 3}),
-            {ok, _} = aetx:check(ContractCreateTx, OnChainTrees,
+            {ok, _} = aetx:process(ContractCreateTx, OnChainTrees,
```
Comment (ThomasArts) [inline apps/aechannel/test/aesc_txs_SUITE.erl]: It looks like you do the same things twice?

File: apps/aechannel/test/aesc_txs_SUITE.erl [hunk 3]
```diff
@@ -3457,7 +3457,7 @@ fp_register_name(Cfg) ->
             ?TEST_LOG("Contract call tx ~p", [CallTx]),
             OnChainTrees = aesc_test_utils:trees(S0),
             TxEnv = tx_env(#{height => 4}),
-            {ok, _} = aetx:check(CallTx, OnChainTrees,
+            {ok, _} = aetx:process(CallTx, OnChainTrees,
```
Comment (ThomasArts) [inline apps/aechannel/test/aesc_txs_SUITE.erl]: Two calls to same function?

File: apps/aechannel/test/aesc_txs_SUITE.erl (hunk 2 above)
Comment (gorillainduction) [reply apps/aechannel/test/aesc_txs_SUITE.erl]: Correct. I'll fix it.

File: apps/aechannel/test/aesc_txs_SUITE.erl (hunk 3 above)
Comment (gorillainduction) [reply apps/aechannel/test/aesc_txs_SUITE.erl]: Correct. I'll fix it.

File: apps/aecore/src/aec_tx_processor.erl [hunk 4]
```diff
@@ -0,0 +1,1443 @@
...
+    MaxTTL = aec_governance:name_claim_max_expiration(),
+    [ inc_account_nonce_op(OwnerPubkey, Nonce)
+    , spend_fee_op(OwnerPubkey, Fee)
+    , name_update_op(OwnerPubkey, NameHash, DeltaTTL, MaxTTL,
+                     ClientTTL, Pointers)
+    ].
+
+-spec contract_create_tx_instructions(pubkey(), amount(), amount(),
+                                      non_neg_integer(), non_neg_integer(),
+                                      non_neg_integer(), non_neg_integer(),
```
Comment (hanssv) [inline apps/aecore/src/aec_tx_processor.erl]: There are specialized types for VM/ABIversion - applies to many places.

File: apps/aecore/src/aec_tx_processor.erl [hunk 5]
```diff
@@ -0,0 +1,1443 @@
...
+                {ok, Store} ->
+                    Contract1 = aect_contracts:set_state(Store, Contract),
+                    S1 = cache_put(contract, Contract1, S),
+                    contract_call_success(InitCall1, GasLimit, S1);
+                {error, _} ->
+                    FailCall0 = aect_call:set_return_value(<<"out_of_gas">>, InitCall),
+                    FailCall  = aect_call:set_return_type(error, FailCall0),
+                    contract_call_fail(FailCall, Fee, RollbackS)
+            end;
+        #{ vm := ?VM_AEVM_SOLIDITY_1} ->
```
Comment (hanssv) [inline apps/aecore/src/aec_tx_processor.erl]: inconsistent whitespacing

File: apps/aecore/src/aec_tx_processor.erl [hunk 6]
```diff
@@ -0,0 +1,1443 @@
...
+        {ok, Hash} ->
+            case aeso_abi:function_name_from_type_hash(Hash, TypeInfo) of
+                {ok, <<"init">>} -> ok;
+                _ -> runtime_error(bad_init_function)
+            end;
+        _Other -> runtime_error(bad_init_function)
+    end.
+
+assert_contract_create_version(ABIVersion, VMVersion, S) ->
+    CTVersion = #{ abi => ABIVersion, vm => VMVersion},
```
Comment (hanssv) [inline apps/aecore/src/aec_tx_processor.erl]: inconsistent whitespacing

File: apps/aens/src/aens_names.erl [hunk 7]
```diff
@@ -59,25 +59,21 @@
...
-    ExpiresBy      = BlockHeight + Expiration,
-    Name           = aens_claim_tx:name(ClaimTx),
-    {ok, NameHash} = aens:get_name_hash(Name),
+-spec new(aens_hash:name_hash(), aec_keys:pubkey(), non_neg_integer()) -> name().
+new(NameHash, OwnerPubkey, AbsoluteTTL) ->
     %% TODO: add assertions on fields, similarily to what is done in aeo_oracles:new/2
-    #name{id         = aec_id:create(name, NameHash),
-          owner_id   = aens_claim_tx:account_id(ClaimTx),
-          expires_by = ExpiresBy,
+     #name{id         = aec_id:create(name, NameHash),
```
Comment (hanssv) [inline apps/aens/src/aens_names.erl]: indentation 5?

File: apps/aeoracle/src/aeo_oracles.erl [hunk 8]
```diff
@@ -83,16 +88,31 @@
...
-               , query_format    = aeo_register_tx:query_format(RTx)
-               , response_format = aeo_register_tx:response_format(RTx)
-               , query_fee       = aeo_register_tx:query_fee(RTx)
-               , ttl             = TTL
-               , abi_version     = aeo_register_tx:abi_version(RTx)
+               , query_format    = QueryFormat
+               , response_format = ResponseFormat
+               , query_fee       = QueryFee
+               , ttl             = AbsoluteTTL
+               , abi_version      = ABIVersion
```
Comment (hanssv) [inline apps/aeoracle/src/aeo_oracles.erl]: misaligned `=`

File: apps/aechannel/src/aesc_channels.erl [hunk 9]
```diff
@@ -216,29 +216,27 @@ pubkey(<<_:?PUB_SIZE/binary>> = InitiatorPubKey, Nonce,
...
+             initiator_id         = aec_id:create(account, InitiatorPubKey),
+             responder_id         = aec_id:create(account, ResponderPubKey),
              channel_amount       = InitiatorAmount + ResponderAmount,
              initiator_amount     = InitiatorAmount,
              responder_amount     = ResponderAmount,
-             channel_reserve      = aesc_create_tx:channel_reserve(ChCTx),
+             channel_reserve      = ReserveAmount,
              delegate_ids         = [aec_id:create(account, D) || D <- DelegatePubkeys],
              state_hash           = StateHash,
-             round                = Round,
```
[the remark this answers is not in the document]
Comment (hanssv) [reply apps/aechannel/src/aesc_channels.erl]: Yes it did.

**Pull request merged**